tooling/tracer/src/lib.rs

```diff
@@ -56,8 +61,15 @@ impl<'a, B: BlackBoxFunctionSolver<FieldElement>> TracingContext<'a, B> {
         initial_witness: WitnessMap<FieldElement>,
         unconstrained_functions: &'a [BrilligBytecode<FieldElement>],
     ) -> Self {
-        let foreign_call_executor =
-            Box::new(DefaultDebugForeignCallExecutor::from_artifact(nargo::PrintOutput::Stdout, debug_artifact));
+        let print_output = Rc::new(RefCell::new(String::new()));
+        let print_output_clone_1 = Rc::clone(&print_output);
+        let print_output_clone_2 = Rc::clone(&print_output);
+        let foreign_call_executor = Box::new(DefaultDebugForeignCallExecutor::from_artifact(
+            nargo::PrintOutput::PrintCallback(Box::new(move |s| {
+                *Rc::clone(&print_output_clone_1).borrow_mut() = s
+            })),
+            debug_artifact,
+        ));
```

Comment:
The pattern to use here is this:

```suggestion
        let foreign_call_executor = Box::new(DefaultDebugForeignCallExecutor::from_artifact(
            nargo::PrintOutput::PrintCallback(Box::new({
                let print_output = Rc::clone(&print_output);
                move |s| *Rc::clone(&print_output).borrow_mut() = s
            })),
            debug_artifact,
        ));
```

Namely: create a short-lived scope; clone and shadow the variable you want to move into a callback.

Reply:
Suggestion applied. thanks!